pvlib/pvsystem.py

```diff
@@ -1794,71 +1846,152 @@ def singlediode(photocurrent, saturation_current, resistance_series,
-
-    params = {'r_sh': resistance_shunt,
-              'r_s': resistance_series,
-              'nNsVth': nNsVth,
-              'i_0': saturation_current,
-              'i_l': photocurrent}
-
-    p_mp, v_mp = _golden_sect_DataFrame(params, 0., v_oc * 1.14, _pwr_optfcn)
-
-    # Invert the Power-Current curve. Find the current where the inverted power
-    # is minimized. This is i_mp. Start the optimization at v_oc/2
-    i_mp = i_from_v(resistance_shunt, resistance_series, nNsVth, v_mp,
-                    saturation_current, photocurrent)
-
-    # Find Ix and Ixx using Lambert W
-    i_x = i_from_v(resistance_shunt, resistance_series, nNsVth, 0.5 * v_oc,
-                   saturation_current, photocurrent)
-
-    i_xx = i_from_v(resistance_shunt, resistance_series, nNsVth,
-                    0.5 * (v_oc + v_mp), saturation_current, photocurrent)
-
-    out = OrderedDict()
-    out['i_sc'] = i_sc
-    out['v_oc'] = v_oc
-    out['i_mp'] = i_mp
-    out['v_mp'] = v_mp
-    out['p_mp'] = p_mp
-    out['i_x'] = i_x
-    out['i_xx'] = i_xx
-
-    # create ivcurve
-    if ivcurve_pnts:
-        ivcurve_v = (np.asarray(v_oc)[..., np.newaxis] *
-                     np.linspace(0, 1, ivcurve_pnts))
+    else:
+        # use singlediode_methods
+        v_from_i_fun = partial(singlediode_methods.bishop88_v_from_i,
+                               method=method.lower())
+        i_from_v_fun = partial(singlediode_methods.bishop88_i_from_v,
+                               method=method.lower())
+        mpp_fun = partial(singlediode_methods.bishop88_mpp,
+                          method=method.lower())
+        args = (photocurrent, saturation_current, resistance_series,
+                resistance_shunt, nNsVth)  # collect args
+        v_oc = v_from_i_fun(0.0, *args)
+        i_mp, v_mp, p_mp = mpp_fun(*args)
+        out = OrderedDict()
+        out['i_sc'] = i_from_v_fun(0.0, *args)
+        out['v_oc'] = v_oc
+        out['i_mp'] = i_mp
+        out['v_mp'] = v_mp
+        out['p_mp'] = p_mp
+        out['i_x'] = i_from_v_fun(v_oc / 2.0, *args)
+        out['i_xx'] = i_from_v_fun((v_oc + v_mp) / 2.0, *args)
+        # calculate the IV curve if requested using bishop88
+        if ivcurve_pnts:
+            vd = v_oc * (
+                    (11.0 - np.logspace(np.log10(11.0), 0.0,
+                                        ivcurve_pnts)) / 10.0
+            )
+            i, v, p = bishop88(vd, *args)
+            out['i'] = i
+            out['v'] = v
+            out['p'] = p
+    return out
 
-        ivcurve_i = i_from_v(resistance_shunt, resistance_series, nNsVth,
-                             ivcurve_v.T, saturation_current, photocurrent).T
 
-        out['v'] = ivcurve_v
-        out['i'] = ivcurve_i
+def mpp(photocurrent, saturation_current, resistance_series, resistance_shunt,
+        nNsVth, method='brentq'):
+    """
+    Given the calculated DeSoto parameters, calculates the maximum power point
+    (MPP).
 
```

Comment:
Add a brief comment why we are setting `brentq` as the default here, instead of `lambertw` which is the default for the `singlediode` function